command/connect/envoy/pipe-bootstrap/connect_envoy_pipe-bootstrap.go

```diff
@@ -0,0 +1,80 @@
+package pipebootstrap
+
+import (
+	"flag"
+	"io"
+	"os"
+	"time"
+
+	"github.com/hashicorp/consul/command/flags"
+	"github.com/mitchellh/cli"
+)
+
+func New(ui cli.Ui) *cmd {
+	c := &cmd{UI: ui}
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+	help  string
+}
+
+func (c *cmd) init() {
+	c.help = flags.Usage(help, c.flags)
+}
+
+func (c *cmd) Run(args []string) int {
+	// This should never be alive for very long. In case bad things happen and
+	// Envoy never starts limit how long we live before just exiting so we can't
+	// accumulate tons of these zombie children.
+	time.AfterFunc(10*time.Second, func() { os.Exit(99) })
+
+	// Read from STDIN, write to the named pipe provided in the only positional arg
+	if len(args) != 1 {
+		c.UI.Error("Expecting named pipe path as argument")
+		return 1
+	}
+
+	// WRONLY is very important here - the open() call will block until there is a
+	// reader (Envoy) if we open it with RDWR though that counts as an opener and
+	// we will just send the data to ourselves as the first opener and so only
+	// valid reader.
+	f, err := os.OpenFile(args[0], os.O_WRONLY|os.O_APPEND, 0700)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	_, err = io.Copy(f, os.Stdin)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	f.Close()
```

Comment:
Since you were doing writes through this file descriptor you should probably explicitly check the error returned from `Close`. I know sometimes for normal files opened for writing some errors only manifest when you close the file descriptor and the rest of the working state is flushed. I know this is a named pipe but it wouldn't hurt to just explicitly check the error.